padertorch/contrib/examples/vae_mnist/mnist_vae_example.py

```diff
@@ -0,0 +1,216 @@
+        prediction = rearrange(outputs['prediction'], 'b c x y -> b (c x y)')
+
+        mse = nn.functional.mse_loss(prediction, images)  # reconstruction error
+
+        q = Normal(loc=outputs['mean'], scale=outputs['std'])  # posterior distribution
+        mean = torch.zeros(outputs['mean'].shape[-1])
+        std = torch.ones(outputs['std'].shape[-1])
+        p = Normal(loc=mean, scale=std)  # prior distribution zero mean and unit variance
+        kld = gaussian_kl_divergence(q=q, p=p)  # regularisation error
+
+        loss = mse + torch.sum(kld)
+
+        return dict(
+            loss=loss,
+        )
+
+
+class VAE(nn.Module):
+    """ A simple Variational Autoencoder"""
+
+    def __init__(self, activation_function, hidden_channels, kernel_size):
+        super().__init__()
+        self.n_layer = 3
+        self.padding = 1
+        self.size_bottleneck = 28  # image size 28*28
+        stride = int(kernel_size/2)
+
+        for idx in range(self.n_layer):
+            self.size_bottleneck = int(
+                (self.size_bottleneck + 2 * self.padding - (kernel_size - 1) - 1) / stride + 1)
+
+        # encoder
+        self.conv_encoder = nn.Sequential(
+            torch.nn.Conv2d(in_channels=1, out_channels=hidden_channels, kernel_size=kernel_size, stride=stride,
+                            padding=self.padding),
+            activation_function(),
+            torch.nn.Conv2d(in_channels=hidden_channels, out_channels=hidden_channels, kernel_size=kernel_size,
+                            stride=stride, padding=self.padding),
+            activation_function(),
+            torch.nn.Conv2d(in_channels=hidden_channels, out_channels=1, kernel_size=kernel_size, stride=stride,
+                            padding=self.padding),
+            activation_function(),
+        )
+        self.linear_1 = torch.nn.Linear(in_features=self.size_bottleneck**2, out_features=1)
+        self.linear_2 = torch.nn.Linear(in_features=self.size_bottleneck**2, out_features=1)
+
+        # decoder
+        self.transp_conv_decoder = nn.Sequential(
+            torch.nn.ConvTranspose2d(in_channels=1, out_channels=hidden_channels, kernel_size=kernel_size,
+                                     stride=stride),
+            activation_function(),
+            torch.nn.ConvTranspose2d(in_channels=hidden_channels, out_channels=hidden_channels,
+                                     kernel_size=kernel_size, stride=stride),
+            activation_function(),
+            torch.nn.ConvTranspose2d(in_channels=hidden_channels, out_channels=1, kernel_size=kernel_size,
+                                     stride=stride),
+        )
+        self.linear_3 = torch.nn.Linear(in_features=1, out_features=self.size_bottleneck**2)
+
+    def encode(self, x):
+        x = self.conv_encoder(x)
+        x = rearrange(x, 'b c x y -> b (c x y)')
+        mu = self.linear_1(x)  # mean value
+        std = 0.5 * self.linear_2(x)  # standard deviation
```

Comment:
Is there a reason for the factor 0.5 here? Also I think it is more common to output `log_var=self.linear_2(x)` with `std = torch.exp(0.5*log_var)`